core/src/por/types.ts

```diff
@@ -0,0 +1,6 @@
+export interface IData {
```

Comment:
If possible, it would be also good if we can reuse the implementation from other files. We don't have to do it now, but we should at least make an issue for it.